### PR TITLE
Adding impl and test tags for requirements

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

This PR adds a smattering of `impl` and `test` tags for requirements in:

* interfaces
* materials
* plugins
* globalFluxInterface

## Why is the change being made?

This is part of the on-going work on ARMI requirements.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
@bdlafleur Just FYI.

File: armi/operators/operator.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ class Operator:
 
     .. note:: The :doc:`/developer/guide` has some additional narrative on this topic.
 
+    .. impl:: The operator package shall expose an ordered list of interfaces, and loop over them in order.
+        :id: I_ARMI_OPERATOR_INTERFACES2
```
Comment (opotowsky) [inline armi/operators/operator.py:77]: I see the `0` but not the `1`, so is this supposed to be `1`?

File: armi/operators/operator.py [hunk 2]
```diff
@@ -650,6 +654,10 @@ def interactAllCoupled(self, coupledIteration):
         This is distinct from loose coupling, which would simply uses the temperature values from the previous timestep
         in the current flux solution. It's also distinct from full coupling where all fields are solved simultaneously.
         ARMI supports tight and loose coupling.
+
+        .. impl:: Physics coupling is driven from Operator.
+            :id: I_ARMI_OPERATOR_PHYSICS2
```
Comment (opotowsky) [inline armi/operators/operator.py:659]: Same as above comment, is this supposed to be `1`?

**Review by opotowsky — CHANGES_REQUESTED** [review]

File: armi/operators/operator.py (hunk 2 above)
Comment (john-science) [reply armi/operators/operator.py]: Fixed!

File: armi/operators/operator.py (hunk 1 above)
Comment (john-science) [reply armi/operators/operator.py]: Fixed!

**Review by opotowsky — APPROVED** [review]
🚀

**Pull request merged**